app/helpers/application_helper/button/transform_vm_button.rb

```diff
@@ -0,0 +1,12 @@
+class ApplicationHelper::Button::TransformVmButton < ApplicationHelper::Button::Basic
+  needs :@record
+
+  def visible?
+    @record.vendor == "vmware"
+  end
+
+  def disabled?
+    # Is there a provider that supports import?
+    EmsInfra.all.select(&:validate_import_vm).empty?
+  end
```

Comment:
What about the `supports`? Seems to be the "simple comparison" or "stubbed return" plus an error message (unsupported reason) are exactly what it does.